scripts/protocol-dts-generator.ts

```diff
@@ -75,153 +66,197 @@ const emitDomain = (domain: P.Domain) => {
     const domainName = toTitleCase(domain.domain)
     emitLine()
     emitDescription(domain.description)
-    emitOpenBlock(`export namespace ${domainName}`)
-    domain.types ? domain.types.forEach(emitType) : null
-    const commandDefs = domain.commands ? domain.commands.map(c => emitCommand(c, domainName)) : []
-    const eventDefs = domain.events ? domain.events.map(e => emitEvent(e, domainName)) : []
+    emitOpenBlock(`export module ${domainName}`)
+    if (domain.types) domain.types.forEach(emitDomainType)
+    if (domain.commands) domain.commands.forEach(emitCommand)
+    if (domain.events) domain.events.forEach(emitEvent)
     emitCloseBlock()
-    const clientDefs = [...commandDefs.map(d => d.client), ...eventDefs.map(d => d.client)]
-    emitInterface(`${domainName}Api`, clientDefs)
-}
-
-const getDomainDef = (domainName: string, $ref:string ) => {
-    return {name: domainName, $ref: `${domainName}${$ref}`}
 }
 
 const getCommentLines = (description: string) => {
     const lines = description
-        .replace(/<code>(.*)<\/code>/g, "'$1'")
         .split(/\r?\n/g)
         .map(line => ` * ${line}`)
-    return [`/**`, ...lines, ,` */`]
+    return [`/**`, ...lines, ` */`]
 }
 
-const emitDescription = (description: string) => {
-    description ? getCommentLines(description).map(l => emitLine(l)) : null
+const emitDescription = (description?: string) => {
+    if (description) getCommentLines(description).map(l => emitLine(l))
 }
 
-const getPropertyDef = (prop: P.ParameterType) => {
-    if ((<any>prop).type == "function") {
-        return `${prop.name}${getPropertyType(prop)}`
-    } else  {
-        return `${prop.name}${prop.optional ? '?' : ''}: ${getPropertyType(prop)}`
-    }
+const getPropertyDef = (prop: P.PropertyType): string => {
+    // Quote key if it has a . in it.
+    const propName = /\./.test(prop.name) ? `'${prop.name}'` : prop.name
```

Comment:
i'm a sucker for str.includes but w/e

Reply:
I'd agree that str.includes would be more readable

Reply:
> I'd agree that str.includes would be more readable

lol ok fine